### ISSUE TITLE
Folders in src/content/docs that are all uppercase will cause failures when referenced in sidebar

### ISSUE THREAD

**Issue by unixben** [withastro/starlight#2487]
### What version of `starlight` are you using?

0.28.3

### What version of `astro` are you using?

4.16.4

### What package manager are you using?

yarn

### What operating system are you using?

Linux

### What browser are you using?

Firefox

### Describe the Bug

In `src/content/docs` I have some folders that are all uppercase:

```
src/content/docs/FAQ
src/content/docs/AU
src/content/docs/NZ
```

When running `yarn dev` or `yarn build`, an `AstroUserError` is thrown:

> The slug "FAQ/help" specified in the Starlight sidebar config does not exist.

If I rename the folder `FAQ` to `frequently-asked-questions`, the page will render without errors.

### Expected behaviour
I should be able to render the sidebar, and pages, with all uppercase directories or names, such as FAQ, RTFM, BYO etc 

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-emifpt

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by apatel369** [comment]
@unixben If you change the slug to lowercase in the astro config it will work.
i.e. slug: 'faq/pagename' instead of slug:'FAQ/pagename'

**Comment by apatel369** [comment]
@unixben I am not sure what you are requesting for. It already has the capability for displaying all uppercase folder name as is in sidebar.

**Comment by unixben** [comment]
Starlight is unable to render pages with uppercase paths, forcing URLs to lowercase.

**Comment by apatel369** [comment]
@unixben It seems to be by design. May be you can close this issue and start a discussion instead?

**Comment by delucis** [comment]
Small update on this: this is a side effect of Astro’s content collections which runs filenames through a slugger to normalize them. One of the things this does is to lowercase slugs. I’m not 100% sure but I would assume that this limitation will go away with Astro v5 as it will support custom loaders, so you could skip such a normalization step and preserve uppercase characters.

That said, our research while building the Astro Docs and Starlight showed that it was often preferable to avoid case sensitivity and default to all lowercase for URLs (and file names). Web servers and operating systems vary a lot in whether they respect differences in case. For example, on macOS and most Linux systems, `foo.md` and `FOO.md` are treated as the same file (you can’t have both). But on Windows these are often considered _different_ files. When collaborating with people across different platforms, dealing with these variations can be a real headache. Similarly, some hosts will normalize URLs to all lowercase (in our case we first noticed this with Netlify). This can be an SEO issue if in your site you have stuff like `<link rel="canonical" href="http://example.com/FOO">`, but the host is redirecting users to `/foo`.

Long story short, we found that sticking to all lowercase avoided a lot of gotchas. Obviously ideally this would still be up to each user if they really need, but wanted to share that information!
